framework/src/modules/chain/dpos/delegates_list.js

```diff
@@ -95,6 +96,35 @@ class DelegatesList extends EventEmitter {
 			round_lt: round,
 		});
 	}
+
+	/**
+	 * Validates if block was forged by correct delegate
+	 *
+	 * @param {Object} block
+	 * @return {Boolean} - `true`
+	 * @throw {Error} Failed to verify slot
+	 */
+	async verifyBlockForger(block) {
+		const currentSlot = this.slots.getSlotNumber(block.timestamp);
+		const round = this.slots.calcRound(block.height);
+		const delegateList = await this.getRoundDelegates(round);
+
+		if (!delegateList.length) {
+			throw new Error(
+				`Failed to verify slot: ${currentSlot} - No delegateList was found`,
+			);
+		}
+
+		// Get delegate key that forged this block
+		const delegatePubKey = delegateList[currentSlot % this.activeDelegates];
+
+		// Verify if forger exists and matches the generatorPublicKey on block
+		if (delegatePubKey && block.generatorPublicKey === delegatePubKey) {
+			return true;
+		}
+
+		throw new Error(`Failed to verify slot: ${currentSlot}`);
```

Comment:
trivial but I'd refactor the code above as below to honor "always error first" rule.

```js
// Verify if forger exists and matches the generatorPublicKey on block
		if (!delegatePubKey || block.generatorPublicKey !== delegatePubKey) {
			throw new Error(`Failed to verify slot: ${currentSlot}`);
		}

return true;
		
```